storage/perfschema/pfs_events.h

```diff
@@ -36,10 +36,10 @@ struct PFS_events
   ulonglong m_end_event_id;
   /** (EVENT_TYPE) */
   enum_event_type m_event_type;
-  /** NESTING_EVENT_ID. */
-  ulonglong m_nesting_event_id;
   /** NESTING_EVENT_TYPE */
   enum_event_type m_nesting_event_type;
+  /** NESTING_EVENT_ID. */
+  ulonglong m_nesting_event_id;
```

Comment:
Wouldn't it become even smaller if we put `ulonglong` first, then pointers, then `int`, then `enum`?

Reply:
Hm. Why? Both `ulonglong` and pointers have the same size and alignment on AMD64. Both `int` and `enum` are 4 bytes long with alignment of 4. So we just don't want to have `ulonglong` after single `int` to avoid padding.

Reply:
enum followed by longlong would usually leave a hole of 4 bytes after enum, so what @svoj says should be correct.

Reply:
I agree with @svoj  on sorting members from largest size (ulonglong) to potentially smaller (pointers, we're not always on AMD64, so they can be 4 bytes long) to enums and ints.  just a good practice to avoid holes in structs

Reply:
I couldn't make it smaller on my machine, but at the very least could you move the `ulonglong` up (rather than down)? Same for pointers in `PFS_events_waits`. We used to have this rule in our coding guidelines, which are unfortunately gone now.

Reply:
OK. I don't mind. PR updated.

Reply:
https://godbolt.org/z/LFDVyT